delfin/drivers/huawei/oceanstor/oceanstor.py

```diff
@@ -692,3 +613,162 @@ def get_capabilities(context):
+                    "ip_address": host['IP']
+                }
+                host_list.append(h)
+
+            return host_list
+
+        except Exception:
+            LOG.error("Failed to get host metrics from OceanStor")
+            raise
+
+    def list_storage_host_groups(self, ctx):
+        try:
+            # Get list of OceanStor host_groups details
+            host_groups = self.client.get_all_host_groups()
+            host_group_list = []
+            for host_group in host_groups:
+                host_g = {
+                    "name": host_group['NAME'],
+                    "description": host_group['DESCRIPTION'],
+                    "storage_id": self.storage_id,
+                    "native_storage_host_group_id": host_group['ID'],
+                }
+                host_group_list.append(host_g)
+
+            return host_group_list
+
+        except Exception:
+            LOG.error("Failed to get host_groups from OceanStor")
+            raise
+
+    def list_port_groups(self, ctx):
+        try:
+            # Get list of OceanStor port_groups details
+            port_groups = self.client.get_all_port_groups()
+            port_group_list = []
+            for port_group in port_groups:
+                port_g = {
+                    "name": port_group['NAME'],
+                    "description": port_group['DESCRIPTION'],
+                    "storage_id": self.storage_id,
+                    "native_port_group_id": port_group['ID'],
+                }
+                port_group_list.append(port_g)
+
+            return port_group_list
+
+        except Exception:
+            LOG.error("Failed to get port_groups from OceanStor")
+            raise
+
+    def list_volume_groups(self, ctx):
+        try:
+            # Get list of OceanStor vol_groups details
+            vol_groups = self.client.get_all_volume_groups()
+            vol_group_list = []
+            for vol_group in vol_groups:
+                vol_g = {
+                    "name": vol_group['NAME'],
+                    "description": vol_group['DESCRIPTION'],
+                    "storage_id": self.storage_id,
+                    "native_volume_group_id": vol_group['ID'],
+                }
+                vol_group_list.append(vol_g)
+
+            return vol_group_list
+
+        except Exception:
+            LOG.error("Failed to get vol_groups from OceanStor")
+            raise
+
+    def list_masking_views(self, ctx):
+        try:
+            # Get list of OceanStor masking view details
+            views = self.client.get_all_mapping_views()
+
+            view_list = []
+            for view in views:
+                v = {
+                    "name": view['NAME'],
+                    "description": view['DESCRIPTION'],
+                    "storage_id": self.storage_id,
+                    "native_masking_view_id": view['ID'],
+                    "native_storage_host_id": view.get('vstoreId'),
```

Comment:
how is this masking view relation? We should have either ( host_group or host ) from initiator side and (volume group or volume ) from LUN side .. I hope we should have different switches to fill masking view of oceanstore. 
1. direct volume to host map
2. mapping though a host group and LUN group

what will be vstore id?

Reply:
Added Host Group, Volume Group and Port Group relations that are supported in OceanStor